ui/shared/utils/constants.js

```diff
@@ -1316,38 +1316,70 @@ export const SV_IN_SILICO_GROUP = 'Structural'
 export const NO_SV_IN_SILICO_GROUPS = [MISSENSE_IN_SILICO_GROUP, CODING_IN_SILICO_GROUP]
 export const SPLICE_AI_FIELD = 'splice_ai'
 
+const PRED_COLOR_MAP = ['green', '#90ee90', 'grey', 'yellow', 'red', '#8b0000']
+
+export const getPredictColor = (value, pathHigher, thresholds) => {
+  let colorIndex = 0
+  const v = pathHigher ? value : -1 * value
+  if (v <= thresholds[1]) {
+    if (v > thresholds[0]) {
+      colorIndex = 1
+    }
+  } else {
+    colorIndex = 5
+    if (v < thresholds[2]) {
+      colorIndex = 2
+    } else if (v < thresholds[3]) {
+      colorIndex = 3
+    } else if (v < thresholds[4]) {
+      colorIndex = 4
+    }
+  }
```

Comment:
This implementation uses the leftmost color if benign thresholds are missing and the right-most color if pathogenic points are missing.